### DISCUSSION TITLE
platform specific style props

### DISCUSSION THREAD

**Discussion by alidcast** [tamagui/tamagui#369] [Ideas]
What's the recommended way to style per platform? 

Have you considered making a style property for this, similar to how pseudo styles or media queries are handled? 

e.g. `<Component platformStyle={{ web: {..}, native: {...}}}` />`

**Comment by BANG88** [comment]
@aleedmanath Tamagui has some constants you can use.

```js
import { useLayoutEffect } from 'react';
export declare const isWeb: boolean;
export declare const isWindowDefined: boolean;
export declare const isServer: boolean;
export declare const isClient: boolean;
export declare const isRSC: any;
export declare const useIsomorphicLayoutEffect: typeof useLayoutEffect;
export declare const isChrome: boolean;
export declare const isWebTouchable: boolean;
export declare const isTouchable: boolean;
//# sourceMappingURL=constants.d.ts.map
```

**Reply by alidcast** [reply]
Yes but these are not as convenient as a component prop would be

**Reply by BANG88** [reply]
Maybe you can implement such a component and use it as you said.
